lang/cs/Org.Apache.REEF.Tests/Functional/IMRU/TestTaskExceptions.cs

```diff
@@ -0,0 +1,251 @@
+                .Build();
+
+            AvroConfigurationSerializer serializer = new AvroConfigurationSerializer();
+            return TangFactory.GetTang().NewConfigurationBuilder(driverConfig)
+                .BindStringNamedParam<TaskConfigurationString>(serializer.ToString(taskConfig))
+                .Build();
+        }
+
+        private IConfiguration GetTaskConfiguration(string taskId, string message)
+        {
+            var taskConfig = TaskConfiguration.ConfigurationModule
+                .Set(TaskConfiguration.Identifier, taskId)
+                .Set(TaskConfiguration.Task, GenericType<ExceptionTask>.Class)
+                .Build();
+
+            var additionalConfig = TangFactory.GetTang().NewConfigurationBuilder()
+                .BindNamedParameter<TaskExceptionMessage, string>(
+                    GenericType<TaskExceptionMessage>.Class, message)
+                .Build();
+
+            return Configurations.Merge(additionalConfig, taskConfig);
+        }
+
+        private sealed class FailedTaskDriver : IObserver<IDriverStarted>, IObserver<IAllocatedEvaluator>,
+            IObserver<IFailedTask>, IObserver<ICompletedTask>
+        {
+            private readonly IEvaluatorRequestor _requestor;
+            private readonly IConfiguration _taskConfiguration;
+
+            [Inject]
+            private FailedTaskDriver(
+                IEvaluatorRequestor requestor,
+                [Parameter(typeof(TaskConfigurationString))] string taskConfigString,
+                AvroConfigurationSerializer avroConfigurationSerializer)
+            {
+                _requestor = requestor;
+                _taskConfiguration = avroConfigurationSerializer.FromString(taskConfigString);
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().Build());
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                value.SubmitTask(_taskConfiguration);
+            }
+
+            /// <summary>
+            /// Verify Exception message and exception type for different task id. 
+            /// </summary>
+            /// <param name="value"></param>
+            public void OnNext(IFailedTask value)
+            {
+                var msg = string.Format(CultureInfo.InvariantCulture,
+                    "In IFailedTask, taskId: {0}, Message {1}, Exception {2}.",
+                    value.Id,
+                    value.Message,
+                    value.AsError().GetType());
+
+                Logger.Log(Level.Info, msg);
+                if (value.Id.Equals(TaskId + 1))
+                {
+                    Assert.Equal(TaskManager.TaskAppError, value.Message);
+                    if (value.AsError() == null || !(value.AsError() is IMRUTaskAppException))
+                    {
+                        throw new Exception("Exception should have been serialized properly.");
```

Comment:
Please add exception type to these error messages (they are otherwise identical and will be hard to distinguish in case of failure).